packages/core/src/commands/serve.cjs

```diff
@@ -38,7 +38,7 @@ const command = async argv => {
 
   spinner.start("Starting off server...");
   // Set port and express server
-  portfinder.basePort = config.port != null ? config.port : argv.port;
+  portfinder.basePort = argv.port !== null ? argv.port : config.port;
```

Comment:
I noticed that for both the serve and the dev command `3000` is passed in as the default in `yargs`. 

I think this would mean that this ternary now has the opposite effect: It will always serve on port 3000 (or any other port you send into the CLI) and always ignore the port specified in your config.

Reply:
Good catch. Tho it's not opposite to what this PR wants, that is to make this behavior consistent between the two commands and take the argv port before config. But yes, given the way it's setup now, the config gets completely ignored (at least in this section).

The config and it's usage isn't really that well realized or though through. It's an idea it made sense at the beginning of developing Mechanic but haven't come back to it or giving it much protagonism (it's not even documented I believe). I propose separating the config issue from this and discuss how to use it and it can be handle in relation to commands.

Reply:
Ok got it! Makes sense, I'm all for merging then and (maybe) addressing the "issue" when we give the role of the config more thought.